### PR TITLE
hv:Replace dynamic memory with static for microcode

### PR DESCRIPTION
**Pull request by mingqiangchi**
Reserve 256KB static memory for microcode

Tracked-On: #861
Signed-off-by: Mingqiang Chi <mingqiang.chi@intel.com>
Acked-by: Anthony Xu <anthony.xu@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
1492<br>1 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/ucode.c&nbsp;&nbsp;function:acrn_update_ucode&nbsp;&nbsp;offset:70:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Cast from pointer to integral type. : (unsigned char to unsigned long): ( uint64_t ) micro_code<br><br>

**Comment by mingqiangchi** [comment]
checked with Junjie, here it is necessary cast from pointer to integral type.

File: hypervisor/arch/x86/guest/ucode.c [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #include <hypervisor.h>
 #include <ucode.h>
 
+#define MICRO_CODE_SIZE_MAX    0x40000U
+static uint8_t micro_code[MICRO_CODE_SIZE_MAX];
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/ucode.c]: I remember ucode update is per core. So, it is possible 2 VCPUs may update ucode in parallel. In this case, we need a lock, right?  Try to see if we can reuse an existing lock..

Comment (mingqiangchi) [reply hypervisor/arch/x86/guest/ucode.c]: can not find an existing lock,  will add anew

**Comment by HuiHuangShi** [comment]
1492<br>1 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/ucode.c&nbsp;&nbsp;function:acrn_update_ucode&nbsp;&nbsp;offset:76:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Cast from pointer to integral type. : (unsigned char to unsigned long): ( uint64_t ) micro_code<br><br>

**Comment by HuiHuangShi** [comment]
1492<br>1 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/ucode.c&nbsp;&nbsp;function:acrn_update_ucode&nbsp;&nbsp;offset:76:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Cast from pointer to integral type. : (unsigned char to unsigned long): ( uint64_t ) micro_code<br><br>

**Comment by ClaudZhang1995** [comment]
Ready to merge

**Pull request merged**